docs/_includes/Refinement.md

```diff
@@ -0,0 +1,130 @@
+The first module is called the _refinement parent_; the second is the
+_refining_ module; the result of combining the two (the original declarations
+and the augmentation instructions) is the _combined_ module or _refinement result_.
+
+Syntactically, the refinement parent is a normal module declaration.
+The refining module declares which module is its refinement parent with the
+`refines` clause:
+```
+module P { // refinement parent
+}
+module M refines P { // refining module
+}
+```
+
+The refinement result is created as follows.
+
+0) The refinement result is a module within the same enclosing module as the
+refining module, has the same name, and in fact replaces the refining module  in their shared scope.
+
+1) All the declarations (including import and export declarations) of the parent are copied into the refinement result.
+These declarations are _not_ re-resolved. That is, the assignment of 
+declarations and types to syntactic names is not changed. The refinement 
+result may exist in a different enclosing module and with a different set of
+imports than the refinement parent, so that if names were reresolved, the 
+result might be different (and possibly not semantically valid).
+Hence it is important that the names not be re-resolved in their new context.
+
+2) All the declarations of the refining module that have different names
+than the declarations in the refinement parent are also copied into the 
+refinement result. Again name and type resolution of the refining module
+happen before this combination takes place.
+
+3) Where declarations in the parent and refinement module have the same name,
+the second refines the first and the combination, a refined declaration, is
+the result placed in the refinement result module, to the exclusion of the
+declarations with the same name from the parent and refinement modules.
+
+The way declarations are combined depends on the kind of declaration;
+the combination rules are described in subsections below.
+
+So that it is clear that refinment is taking place, refining declarations
+have some syntactic indicator that they are refining some parent declaration.
+Typically this is the presence of a `...` token.
+
+## 22.47. Export set declarations
+
+A refining export set declaration begins with the syntax
+```grammar
+"export" Ident "..." 
+```
+but otherwise contains the same `provides`, `reveals` and `extends` sections,
+with the ellipsis indicating that it is a refining declaration.
+
+The combined result declaration has the same name as the two input declarations and the unions of names from each of the `provides`, `reveals`, and `extends`
+sections, respectively.
+
+An unnamed export set declaration from the parent is copied into the result
+module with the name of the parent module. The result module has a default
+export set according to the general rules for export sets, after all of 
+the result module's export set declarations have been assembled.
```

Comment:
I think this is the first use of the word "assemble" here. It's a pretty good word. Should it be used instead of the word "combined"? (If not, then I think it's better to change this "assembled" to "combined".)

Reply:
Used assembled and edited accordingly